docs/specs/clients/sign/wallet-authentication.md

```diff
@@ -0,0 +1,161 @@
+
+A subscribes to messages on response topic.
+
+B receives request on A-B pairing.
+
+B constructs message to be signed from request and signs it.
+
+B generates keyPair Y and generates shared symKey R.
+
+B encrypts response with symKey R as type 1 envelope.
+
+B sends response on response topic.
+
+A receives response and validates signature.
+
+If signature is valid, then user is authenticated.
+
+## Capabilities
+
+Applications can request consent from users to perform actions on their behalf. These actions are included in the authentication request in the form of ReCaps.
+WalletConnect's implementation follows closely the [ERC-5573 specs](https://eips.ethereum.org/EIPS/eip-5573). ReCaps are also used as a form of handshake to establish Sign session.
+
+### Overview
+
+ReCaps capabilities are included in the auth request in the `resources` array. They are prefixed with `urn:recap:` and are `base64` encoded.
+Authentication message with recaps would look like
+
+```typescript
+http://example.com wants you to sign in with your Ethereum account:
+0x3613699A6c5D8BC97a08805876c8005543125F09
+
+I further authorize the stated URI to perform the following actions on my behalf: (1) 'request': 'eth_signTypedData_v4', 'personal_sign' for 'eip155'.
+
+URI: https://example.com
+Version: 1
+Chain ID: 1
+Nonce: 1
+Issued At: 2024-02-19T09:29:21.394Z
+Resources:
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InJlcXVlc3QvZXRoX3NpZ25UeXBlZERhdGFfdjQiOlt7fV0sInJlcXVlc3QvcGVyc29uYWxfc2lnbiI6W3t9XX19fQ==
+```
+
+The recap in the message
+
+```
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InJlcXVlc3QvZXRoX3NpZ25UeXBlZERhdGFfdjQiOlt7fV0sInJlcXVlc3QvcGVyc29uYWxfc2lnbiI6W3t9XX19fQ==
+```
+
+after removing the prefix (`urn:recap:`) and decoding would result in
+
+```
+{
+   "att":{
+      "eip155":{
+         "request/eth_signTypedData_v4":[{}],
+         "request/personal_sign":[{}]
+      }
+   }
+}
+```
+
+This ReCap Object is generated by the requesting application and is used to build the statement
+
+```
+I further authorize the stated URI to perform the following actions on my behalf: (1) 'request': 'eth_signTypedData_v4', 'personal_sign' for 'eip155'.
+```
+
+Any data following above format & [ERC-5573 specs](https://eips.ethereum.org/EIPS/eip-5573) can be encoded and requested as part of Authentication request, even multiple recaps.
+
+```
+http://example.com wants you to sign in with your Ethereum account:
+0x3613699A6c5D8BC97a08805876c8005543125F09
+
+I further authorize the stated URI to perform the following actions on my behalf: (1) 'request': 'eth_signTypedData_v4', 'personal_sign' for 'eip155'. (2) 'push': 'messages', 'notification' for 'eip155'. (3) 'receive': 'messages', 'notification' for 'eip155'.
+
+URI: https://example.com
+Version: 1
+Chain ID: 1
+Nonce: 1
+Issued At: 2024-02-19T09:29:21.394Z
+Resources:
+- urn:recap:eyJhdHQiOnsiZWlwMTU1Ijp7InJlcXVlc3QvZXRoX3NpZ25UeXBlZERhdGFfdjQiOlt7fV0sInJlcXVlc3QvcGVyc29uYWxfc2lnbiI6W3t9XX19fQ==
+- https://example.com
```

Comment:
having a bare "example.com" here is a little confusing, better to give a realistic example of what example might encode as a URI as an "additional authorization" not covered by the SIWE itself.
```suggestion
- https://example.com/storage/0x3613699A6c5D8BC97a08805876c8005543125F09
```